src/bootstrap/src/utils/exec.rs

```diff
@@ -780,6 +781,8 @@ impl ExecutionContext {
         if !command.run_in_dry_run && self.dry_run() {
             return None;
         }
+        let start_time = Instant::now();
+        let cache_key = command.cache_key();
```

Comment:
We should also rename the `cache_key` method to `fingerprint`, now that I see it :)